### PR TITLE
fix(gatsby-plugin-image): Use bare GATSBY___IMAGE global

### LINKED ISSUE #30093: [gatsby-plugin-image] causes Uncaught ReferenceError: `global` is not defined
## Overview

`gatsby-plugin-image` isn't playing nice with other plugins like `gatsby-plugin-use-dark-mode` that require the `global` variable.

I might be wrong here but,
based on a minimal reproduction and some debugging,
I think that the issue lies with the plugin's **Gatsby Node API** calls that are affecting
the **webpack configuration**.

## Reproduction

You can follow the instructions below or refer [this example on CodeSandbox](https://codesandbox.io/s/use-dark-mode-wgatsby-v3-mb3wf):

- Start with `gatsby-starter-default`.
- Add these packages:

  ```bash
  yarn add gatsby-plugin-use-dark-mode use-dark-mode
  ```

- Upgrade packages (optional)

  ```bash
  yarn upgrade
  ```

- Add `gatsby-plugin-use-dark-mode` to the list of plugins.
- Invoke the `useDarkMode` hook in the `Layout` component, destructure `value` and wrap it in a `div`.

### Actual

The development server throws no relevant warnings or errors in the terminal.

The browser console logs the following error message:

```
Uncaught ReferenceError: global is not defined
    at d (use-dark-mode.m.js:1)
```

The same can be seen in the following screenshot:

Uncaught RefernceError: global is not defined

### Expected

This error should not exist and the site should run smoothly.

## Environment

<details>
<summary>

Running `gatsby info` on the reproduction.

</summary>

```
  System:
    OS: Linux 5.4 Manjaro Linux
    CPU: (8) x64 Intel(R) Core(TM) i7-4700HQ CPU @ 2.40GHz
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 14.15.4 - ~/.nvm/versions/node/v14.15.4/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v14.15.4/bin/yarn
    npm: 6.14.10 - ~/.nvm/versions/node/v14.15.4/bin/npm
  Languages:
    Python: 3.9.1 - /usr/bin/python
  npmPackages:
    gatsby: ^3.0.1 => 3.0.3
    gatsby-plugin-gatsby-cloud: ^2.0.0 => 2.0.0
    gatsby-plugin-image: ^1.0.0 => 1.0.0
    gatsby-plugin-manifest: ^3.0.0 => 3.0.0
    gatsby-plugin-offline: ^4.0.0 => 4.0.0
    gatsby-plugin-react-helmet: ^4.0.0 => 4.0.0
    gatsby-plugin-sharp: ^3.0.0 => 3.0.0
    gatsby-plugin-use-dark-mode: ^1.3.0 => 1.3.0
    gatsby-source-filesystem: ^3.0.0 => 3.0.0
    gatsby-transformer-sharp: ^3.0.0 => 3.0.0
  npmGlobalPackages:
    gatsby-cli: 2.19.1
```

</details>

## Conclusion

As mentioned previously, I think there issue lies with the changing of the Webpack configuration.
The polyfill for `global` seems to be affected, since the browser says it is not `defined`.

### LINKED ISSUE #30609: Error in node_modules: global is not defined
## Description

Since upgrading to Gatsby V3 and using the new **gatsby-plugin-image**, some of my node packages throw an exception in the browser with `global is not defined`

The following ones confirmed, but others have had the same issue:
[voca](https://github.com/panzerdp/voca)
[reaptcha](https://github.com/sarneeh/reaptcha)

### Steps to reproduce

Create a new gatsby site with **gatsby-starter-default**
Update **gatsby**, **gatsby-plugin-image** and **gatsby-plugin-sharp** to the latest versions

At time of writing:
```
gatsby: ^3.2.0 => 3.2.0
gatsby-plugin-image: ^1.2.0 => 1.2.0
gatsby-plugin-sharp: ^3.2.0 => 3.2.0
```

Run **npm install voca**

In index.js add:
```
import voca from "voca"
console.log(voca.kebabCase("goodbye blue sky"))
```

Run **npm run develop**
Open localhost:8000 and the error should appear

**I can create a reproduction if necessary** but the example is pretty straghtforward 

### Expected result

Packages should work as they did in Gatsby V2

### Actual result

Exception is thrown in the browser, usually to the tune of

`Error in function eval in: ./node_modules/xxxx
global is not defined.`

I suspect this is due to a change in webpack v5, where somehow the target messes up the globals leaving the package in state where it thinks it's running in Node, but is in fact not.

It is especially strange in vocajs, as the exception is thrown in this line:

```
    if (typeof global === 'object' && global.Object === Object) {
        // NodeJS global object
 >    globalObject$1 = global;
```
I have no idea how it even gets into that if statement.

Any leads would be appreciated as I would desperately like to use the new **gatsby-plugin-image** in our project

### Environment

System:
    OS: Linux 5.8 Ubuntu 20.04.2 LTS (Focal Fossa)
    CPU: (4) x64 Intel(R) Core(TM) i5-7300U CPU @ 2.60GHz
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 14.16.0 - ~/.nvm/versions/node/v14.16.0/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v14.16.0/bin/yarn
    npm: 6.14.11 - ~/.nvm/versions/node/v14.16.0/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 89.0.4389.114
    Firefox: 87.0
  npmPackages:
    gatsby: ^3.0.4 => 3.0.4
    gatsby-background-image: ^1.4.1 => 1.4.1
    gatsby-cli: ^3.0.0 => 3.0.0
    gatsby-image: ^3.0.0 => 3.0.0
    gatsby-plugin-disqus: ^1.2.0 => 1.2.2
    gatsby-plugin-google-analytics: ^3.0.0 => 3.0.0
    gatsby-plugin-google-gtag: ^3.0.0 => 3.0.0
    gatsby-plugin-manifest: ^3.0.0 => 3.0.0
    gatsby-plugin-nprogress: ^3.0.0 => 3.0.0
    gatsby-plugin-offline: ^4.0.0 => 4.0.0
    gatsby-plugin-preconnect: ^1.2.0 => 1.2.0
    gatsby-plugin-react-helmet: ^4.0.0 => 4.0.0
    gatsby-plugin-remove-console: 0.0.2 => 0.0.2
    gatsby-plugin-sharp: ^3.0.1 => 3.0.1
    gatsby-plugin-sitemap: ^3.0.0 => 3.0.0
    gatsby-plugin-styled-components: ^4.0.0 => 4.0.0
    gatsby-plugin-use-query-params: ^1.0.1 => 1.0.1
    gatsby-plugin-webpack-bundle-analyser-v2: ^1.1.21 => 1.1.21
    gatsby-source-filesystem: ^3.0.0 => 3.0.0
    gatsby-source-graphql: ^3.0.0 => 3.0.0
    gatsby-source-woocommerce: ^0.2.0 => 0.2.0
    gatsby-source-wordpress: ^5.0.0 => 5.0.0
    gatsby-transformer-json: ^3.0.0 => 3.0.0
    gatsby-transformer-sharp: ^3.0.0 => 3.0.0
  npmGlobalPackages:
    gatsby-cli: 3.2.0

### PR DESCRIPTION
**Pull request by ascorbic**
The image plugin currently defines `global.GATSBY___IMAGE` to detect when the plugin is installed, which causes problems with thrid-party plugins. This changes it to just create a bare global `GATSBY___IMAGE`, and updates types to match. 

Thanks @kelvindecosta for his PR, but there were some issues in the approach there so I have opened this instead.

Fixes #30093, fixes #30094, fixes #30456, fixes #30609

### CODE REVIEW

**Comment by kelvindecosta** [comment]
Thank you for the feedback!

I'll admit that my fix was rather crude, and I'm glad that this PR solves the underlying issue!

Thanks again!

**Review by pieh — APPROVED** [review]
Make sense, thanks!

File: packages/gatsby-plugin-image/src/components/__tests__/gatsby-image.browser.tsx [hunk 1]
```diff
@@ -19,8 +19,9 @@ describe(`GatsbyImage browser`, () => {
 
   beforeEach(() => {
     console.warn = jest.fn()
-    ;(global as GlobalOverride).SERVER = true
```
Comment (LekoArts) [inline packages/gatsby-plugin-image/src/components/__tests__/gatsby-image.browser.tsx:22]: Is the GlobalOverride from above now unused/obsolete?

Comment (ascorbic) [reply packages/gatsby-plugin-image/src/components/__tests__/gatsby-image.browser.tsx:22]: Yes! Good spot. Removed.

**Review by LekoArts — APPROVED** [review]
🌮

**Comment by LekoArts** [comment]
Published in `gatsby-plugin-image@1.2.1`

**Pull request merged**